pkg/controller/operators/catalog/operator.go

```diff
@@ -1424,49 +1426,52 @@ func (o *Operator) syncInstallPlans(obj interface{}) (syncError error) {
 }
 
 type installPlanTransitioner interface {
-	ResolvePlan(*v1alpha1.InstallPlan) error
 	ExecutePlan(*v1alpha1.InstallPlan) error
 }
 
 var _ installPlanTransitioner = &Operator{}
 
-func transitionInstallPlanState(log *logrus.Logger, transitioner installPlanTransitioner, in v1alpha1.InstallPlan, now metav1.Time) (*v1alpha1.InstallPlan, error) {
+func transitionInstallPlanState(log logrus.FieldLogger, transitioner installPlanTransitioner, in v1alpha1.InstallPlan, now metav1.Time, timeout time.Duration) (*v1alpha1.InstallPlan, error) {
 	out := in.DeepCopy()
 
 	switch in.Status.Phase {
 	case v1alpha1.InstallPlanPhaseRequiresApproval:
 		if out.Spec.Approved {
-			log.Debugf("approved, setting to %s", v1alpha1.InstallPlanPhasePlanning)
 			out.Status.Phase = v1alpha1.InstallPlanPhaseInstalling
+			out.Status.Message = ""
+			log.Debugf("approved, setting to %s", out.Status.Phase)
 		} else {
 			log.Debug("not approved, skipping sync")
 		}
 		return out, nil
 
 	case v1alpha1.InstallPlanPhaseInstalling:
+		if out.Status.StartTime == nil {
+			out.Status.StartTime = &now
+		}
 		log.Debug("attempting to install")
 		if err := transitioner.ExecutePlan(out); err != nil {
-			out.Status.SetCondition(v1alpha1.ConditionFailed(v1alpha1.InstallPlanInstalled,
-				v1alpha1.InstallPlanReasonComponentFailed, err.Error(), &now))
-			out.Status.Phase = v1alpha1.InstallPlanPhaseFailed
+			if now.Sub(out.Status.StartTime.Time) >= timeout {
+				out.Status.SetCondition(v1alpha1.ConditionFailed(v1alpha1.InstallPlanInstalled,
+					v1alpha1.InstallPlanReasonComponentFailed, err.Error(), &now))
```

Comment:
Do we want to mention that the error failed even with retries in the status?

For example, say installplan failed due to a transient connection error with the api-server (that persisted for some amount of time). Is it more helpful to say "installplan failed due to connection error with api-server: timeout <interval>" versus just "installplan failed due to connection error"? 

This leaks to the user that we retried the plan, but maybe that's what we would want in cases like this?

Reply:
It actually does do this!

Reply:
sweet